vendor/src/github.com/docker/go-connections/nat/nat.go

```diff
@@ -146,9 +146,31 @@ func ParsePortSpecs(ports []string) (map[Port]struct{}, map[Port][]PortBinding,
 			rawPort = fmt.Sprintf(":%s", rawPort)
 		}
 
-		parts, err := PartParser(portSpecTemplate, rawPort)
-		if err != nil {
-			return nil, nil, err
+		parts := make(map[string]string)
+		if rawPort != "" && rawPort[0] == '[' {
```

Comment:
> Even if we don’t need them, I think it makes sense to use them. Using square brackets for separating IPv6 addresses from ports is what all other programs do, so there’s a strong argument for consistency.

Sticking to ipv6 literal notation is very good idea. It's too easy to get lost in plethora of colons without square brackets (not to mention that the number of colons in ipv6 can vary).
https://www.ietf.org/rfc/rfc2732.txt